src/txn/txn.c

```diff
@@ -401,6 +402,24 @@ __wt_txn_commit(WT_SESSION_IMPL *session, const char *cfg[])
 	if (!F_ISSET(txn, TXN_RUNNING))
 		WT_RET_MSG(session, EINVAL, "No transaction is active");
 
+	/*
+	 * The default sync setting is inherited from the connection, but can
+	 * be overridden by an explicit "sync" setting for this transaction.
+	 *
+	 * !!! This is an unusual use of the config code: the "default" value
+	 * we pass in is inherited from the connection.  If flush is not set in
+	 * the connection-wide flag and not overridden here, we end up clearing
+	 * all flags.
+	 */
+	WT_RET(__wt_config_gets_def(session, cfg, "sync",
+	    FLD_ISSET(txn->txn_logsync, WT_LOG_FLUSH) ? 1 : 0, &cval));
```

Comment:
I kind of understand what you mean here, but I don't think this will work as written.

Passing a non-constant default value to `__wt_config_gets_def` was a weird thing to do for `begin_transaction` and I'd prefer to get rid of it rather than add more instances.  Regardless, if the application doesn't supply a `"sync"` config, this call will set `cval.str` to the empty string, which would fall through to your `else` case and set `txn->txn_logsync = 0`.

You could pass in an invalid value here, like `UINT64_MAX`, then check for that value to indicate that nothing was passed in -- that might be closer to the intent?